library/std/src/panicking.rs

```diff
@@ -658,9 +658,11 @@ pub const fn begin_panic<M: Any + Send>(msg: M) -> ! {
 
     let loc = Location::caller();
     return crate::sys_common::backtrace::__rust_end_short_backtrace(move || {
+        let message =
+            *(&msg as &dyn Any).downcast_ref::<&'static str>().unwrap_or(&"<non-str payload>");
```

Comment:
> `"<non-str payload>"`

The default panic hook prints `"Box<dyn Any>"` in that case. I'm not particularly attached to that specific message, but we should probably keep those the same.

https://github.com/rust-lang/rust/blob/8769c269895bfafca92fd7cba9475425f8d09ba0/library/std/src/panicking.rs#L264

Reply:
Or, even better: the default panic hook should not do any downcasting and instead use .message(), now that it works in all cases.